File: solutions/zig/space-age/1/space_age.zig
pub const Planet = enum {
    earth,
    mercury,
    venus,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    const one_year_in_seconds: f64 = 31_557_600;

    pub fn age(self: Planet, seconds: usize) f64 {
        const factor: f64 = switch (self) {
            .earth => 1.0,
            .mercury => 0.2408467,
            .venus => 0.61519726,
            .mars => 1.8808158,
            .jupiter => 11.862615,
            .saturn => 29.447498,
            .uranus => 84.016846,
            .neptune => 164.79132,
        };
        return @as(f64, @floatFromInt(seconds)) / one_year_in_seconds / factor;
    }
};

File: solutions/haskell/space-age/1/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

ageOn :: Planet -> Float -> Float
ageOn Earth seconds = seconds * 1.0 / (60 * 60 * 24 * 365.25)
ageOn Mercury seconds = ageOn Earth seconds / 0.2408467
ageOn Venus seconds = ageOn Earth seconds / 0.61519726
ageOn Mars seconds = ageOn Earth seconds / 1.8808158
ageOn Jupiter seconds = ageOn Earth seconds / 11.862615
ageOn Saturn seconds = ageOn Earth seconds / 29.447498
ageOn Uranus seconds = ageOn Earth seconds / 84.016846
ageOn Neptune seconds = ageOn Earth seconds / 164.79132
